frontend/src/MuiForms/TagSelectorBeta.tsx

```diff
@@ -0,0 +1,78 @@
+import { Autocomplete, Box, Chip, Stack, TextField, Typography } from '@mui/material'
+import { useTheme } from '@mui/material/styles'
+
+export default function TagSelector(props: any) {
+  const { onChange, value: currentValue, label, formContext } = props
+
+  const _onChange = (_event: React.SyntheticEvent<Element, Event>, newValues: any) => {
+    onChange(newValues.map((value) => value))
+  }
+
+  const theme = useTheme()
+
+  return (
+    <>
+      {formContext.editMode && (
+        <>
+          <Typography sx={{ fontWeight: 'bold' }}>{label}</Typography>
+          <Autocomplete
+            multiple
+            isOptionEqualToValue={(option: any, value: any) => option === value}
+            value={currentValue || ''}
+            onChange={_onChange}
+            options={[]}
+            freeSolo
+            renderTags={(value: readonly string[], getTagProps) =>
```

Comment:
We shouldn't be mutating parameters ever, so specifying `value` as `readonly` is unnecessary. I believe this is already covered by our eslint rules also.